### PR TITLE
Bt wifi buffer corruption

### PR DESCRIPTION
**Pull request by peterharperuk**
Fix BT+Wifi buffer corruption issue.

Change from Infineon to fix the BT buffer corruption which occurs when
making heavy use of BT and WIFI at the same time.

### CODE REVIEW

**Comment by peterharperuk** [comment]
Also want this https://github.com/georgerobotics/cyw43-driver/pull/77

File: src/rp2_common/pico_cyw43_driver/cybt_shared_bus/cybt_shared_bus_driver.c [hunk 1]
```diff
@@ -559,10 +554,19 @@ cybt_result_t cybt_get_bt_buf_index(cybt_fw_membuf_index_t *p_buf_index) {
 static cybt_result_t cybt_reg_write(uint32_t reg_addr, uint32_t value) {
     cybt_debug("cybt_reg_write 0x%08lx 0x%08lx\n", reg_addr, value);
     cyw43_ll_write_backplane_reg(cyw43_ll, reg_addr, value);
+    if (reg_addr == HOST_CTRL_REG_ADDR)
+    {
+        host_ctrl_cache_reg = value;
+    }
```
Comment (dpgeorge) [inline src/rp2_common/pico_cyw43_driver/cybt_shared_bus/cybt_shared_bus_driver.c:560]: Maybe format it with the open curly brace on the same line as the `if`, to match surrounding code?

File: src/rp2_common/pico_cyw43_driver/cybt_shared_bus/cybt_shared_bus_driver.c [hunk 2]
```diff
@@ -592,7 +596,11 @@ static void dump_bytes(const uint8_t *bptr, uint32_t len) {
 static cybt_result_t cybt_mem_write(uint32_t mem_addr, const uint8_t *p_data, uint32_t data_len) {
     cybt_debug("cybt_mem_write addr 0x%08lx len %ld\n", mem_addr, data_len);
     do {
-        uint32_t transfer_size = (data_len > MAX_BLOCK_SIZE) ? MAX_BLOCK_SIZE : data_len;
+        uint32_t transfer_size = (data_len > CYW43_BUS_MAX_BLOCK_SIZE) ? CYW43_BUS_MAX_BLOCK_SIZE : data_len;
+        if ((mem_addr & 0xFFF) + transfer_size > 0x1000)
+        {
```
Comment (dpgeorge) [inline src/rp2_common/pico_cyw43_driver/cybt_shared_bus/cybt_shared_bus_driver.c:601]: Code style?

File: src/rp2_common/pico_cyw43_driver/cybt_shared_bus/cybt_shared_bus_driver.c [hunk 3]
```diff
@@ -607,8 +615,11 @@ static cybt_result_t cybt_mem_read(uint32_t mem_addr, uint8_t *p_data, uint32_t
     assert(data_len >= 4);
     cybt_debug("cybt_mem_read addr 0x%08lx len %ld\n", mem_addr, data_len);
     do {
-        uint32_t transfer_size = (data_len > MAX_BLOCK_SIZE) ? MAX_BLOCK_SIZE : data_len;
-        /* this limitation from BT, we need to read twice when spi clock setting is more than 25MHz */
+        uint32_t transfer_size = (data_len > CYW43_BUS_MAX_BLOCK_SIZE) ? CYW43_BUS_MAX_BLOCK_SIZE : data_len;
+        if ((mem_addr & 0xFFF) + transfer_size > 0x1000)
+        {
```
Comment (dpgeorge) [inline src/rp2_common/pico_cyw43_driver/cybt_shared_bus/cybt_shared_bus_driver.c:620]: Code style?

**Comment by peterharperuk** [comment]
@dpgeorge Hopefully the code formatting issues are fixed now. Thanks a lot for your help with this issue!

**Comment by dpgeorge** [comment]
The cyw43-driver fix has been merged into that repo.  There is no tag/release for it yet, but you can just point to the latest commit on the main branch.

**Review by kilograham — APPROVED** [review]

**Pull request merged**